src/Exceptions/CannotCreateData.php

```diff
@@ -42,6 +41,6 @@ public static function constructorMissingParameters(
                 ->map(fn (DataProperty|DataParameter $parameter) => $parameter->name)
                 ->join(', ')}.";
 
-        return new self($message, previous: $previous);
```

Comment:
Looks good to me, except this deletion

Reply:
Previously `ArgumentCountError` was thrown, but now hence we do the checks based on the data class properties, I had to remove that because there is no previous error.

Reply:
I must have been squinting and saw my suggested change as the code. Have a look at the suggestion, basically, keep the try catch but do the check only if it throws, like this you let php handle the argument validation